lib/main.dart

```diff
@@ -106,20 +109,23 @@ class _SplashScreenState extends State<SplashScreen> with SingleTickerProviderSt
     );
   }
 
+// Carlo
   @override
   void dispose() {
     _fadeController.dispose();
     super.dispose();
   }
 }
 
+// Carlo
 class HomeScreen extends StatefulWidget {
   const HomeScreen({Key? key}) : super(key: key);
 
   @override
   _HomeScreenState createState() => _HomeScreenState();
 }
 
+// Carlo
```

Comment:
For the different voids, possibly add else statements for where the index is out of bounds (error handling).